src/algebra/squarefree.lean

```diff
@@ -83,6 +86,33 @@ begin
   norm_cast,
 end
 
+end comm_monoid
+
+section cancel_comm_monoid_with_zero
+
+variables [cancel_comm_monoid_with_zero R] [decidable_rel (has_dvd.dvd : R → R → Prop)]
+  [wf_dvd_monoid R]
+
+lemma finite_prime_left {a b : R} (ha : prime a) (hb : b ≠ 0) :
```

Comment:
linter's right in all the cases about this - you only need to have it when the statement depends on that decidability, not in general. when it's not in the statement, we're allowed to have use `classical`, but if it is in the statement then you can get `rw` issues that look like "there is no `X y` in the expression `X y = b", because the first one is `@X (classical.dec_eq) y` whilst the second one is `@X my_dec_eq y`. does that make sense? sorry if I wasn't clear before

Reply:
Oh right, that makes sense, I wasn't aware of that issue. I've made the corresponding changes!

Reply:
Thanks for the explanation ;-)